irr/src/CGLTFMeshFileLoader.h

```diff
@@ -0,0 +1,145 @@
+#ifndef __C_GLTF_MESH_FILE_LOADER_INCLUDED__
+#define __C_GLTF_MESH_FILE_LOADER_INCLUDED__
+
+#include "ISkinnedMesh.h"
+#include "IMeshLoader.h"
+#include "IReadFile.h"
+#include "irrTypes.h"
+#include "path.h"
+#include "S3DVertex.h"
+#include "vector2d.h"
+#include "vector3d.h"
+
+#include <tiniergltf.hpp>
+
+#include <cstddef>
+#include <vector>
+
+namespace irr
+{
+
+namespace scene
+{
+
+class CGLTFMeshFileLoader : public IMeshLoader
+{
+public:
+	CGLTFMeshFileLoader() noexcept;
+
+	bool isALoadableFileExtension(const io::path& filename) const override;
+
+	IAnimatedMesh* createMesh(io::IReadFile* file) override;
+
+private:
+	class BufferOffset
+	{
+	public:
+		BufferOffset(const std::vector<unsigned char>& buf,
+				const std::size_t offset);
+
+		BufferOffset(const BufferOffset& other,
+				const std::size_t fromOffset);
+
+		unsigned char at(const std::size_t fromOffset) const;
+	private:
+		const std::vector<unsigned char>& m_buf;
+		std::size_t m_offset;
+	};
+
+	class MeshExtractor {
```

Comment:
Class violates rule of 5.

Reply:
I was wrong. These aren't the copy/move constructors. These are custom constructors.